napari/layers/tracks/tracks.py

```diff
@@ -0,0 +1,520 @@
+        coords = np.array(self.coordinates)
+        return self._manager.get_value(coords)
+
+    def _update_thumbnail(self):
+        """Update thumbnail with current points and colors."""
+        colormapped = np.zeros(self._thumbnail_shape)
+        colormapped[..., 3] = 1
+        # return
+        if self._view_data is not None and self.track_colors is not None:
+            de = self._extent_data
+            min_vals = [de[0, i] for i in self.dims.displayed]
+            shape = np.ceil(
+                [de[1, i] - de[0, i] + 1 for i in self.dims.displayed]
+            ).astype(int)
+            zoom_factor = np.divide(
+                self._thumbnail_shape[:2], shape[-2:]
+            ).min()
+            if len(self._view_data) > self._max_tracks_thumbnail:
+                thumbnail_indices = np.random.randint(
+                    0, len(self._view_data), self._max_tracks_thumbnail
+                )
+                points = self._view_data[thumbnail_indices]
+            else:
+                points = self._view_data
+                thumbnail_indices = range(len(self._view_data))
+
+            # get the track coords here
+            coords = np.floor(
+                (points[:, :2] - min_vals[1:] + 0.5) * zoom_factor
+            ).astype(int)
+            coords = np.clip(
+                coords, 0, np.subtract(self._thumbnail_shape[:2], 1)
+            )
+
+            # modulate track colors as per colormap/current_time
+            colors = self.track_colors[thumbnail_indices]
+            times = self.track_times[thumbnail_indices]
+            alpha = (self.current_time - times) / self.tail_length
+            alpha[times > self.current_time] = 1.0
+            colors[:, -1] = np.clip(1.0 - alpha, 0.0, 1.0)
+            colormapped[coords[:, 1], coords[:, 0]] = colors
+
+        colormapped[..., 3] *= self.opacity
+        self.thumbnail = colormapped
+
+    @property
+    def _view_data(self):
+        """ return a view of the data """
+        return self._pad_display_data(self._manager.track_vertices)
+
+    @property
+    def _view_graph(self):
+        """ return a view of the graph """
+        return self._pad_display_data(self._manager.graph_vertices)
+
+    def _pad_display_data(self, vertices):
+        """ pad display data when moving between 2d and 3d
+
+        NOTES:
+            2d data is transposed yx
+            3d data is zyxt
+
+        """
+        if vertices is None:
+            return
+
+        data = vertices[:, self.dims.displayed]
+        # if we're only displaying two dimensions, then pad the display dim
+        # with zeros
+        if self.dims.ndisplay == 2:
+            data = np.pad(data, ((0, 0), (0, 1)), 'constant')
+            data = data[:, (1, 0, 2)]  # y, x, z -> x, y, z
+        else:
+            data = data[:, (2, 1, 0)]  # z, y, x -> x, y, z
+
+        # return the data + 0.5 pixel offset
+        return data + 0.5
```

Comment:
@sofroniewn if #1660 merges, does it mean we get rid of this 0.5?

Reply:
Yes, now that #1660 is merged this `0.5` can be dropped